### PR TITLE
[search-in-workspace] fix incorrect file-icon in 'search-in-workspace'

### LINKED ISSUE #3597: [search-in-workspace] incorrect file-icon displayed
`search-in-workspace` displays wrong file-icon for files which have no extension (treats them instead as directories). Propose to instead use the `FileStat` when getting the `icon` from the `labelProvider` like other instances instead of `SearchInWorkspaceResult.file`.

screen shot 2018-11-22 at 5 27 20 pm

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Fixes #3597

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx [hunk 1]
```diff
@@ -187,7 +189,8 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
                     }
                 } else {
                     const children: SearchInWorkspaceResultLineNode[] = [];
-                    const icon = await this.labelProvider.getIcon(new URI(result.file));
+                    const fileStat = await this.fileSystem.getFileStat(result.file);
+                    const icon = (fileStat) ? await this.labelProvider.getIcon(fileStat) : undefined;
```
Comment (simark) [inline packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx]: Nit: unnecessary parentheses arounf `fileStat`?

Comment (simark) [reply packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx]: Just wondering, what is displayed if `icon` ends up undefined?

Comment (vince-fugnitto) [reply packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx]: > Just wondering, what is displayed if `icon` ends up undefined?

if the icon is `undefined` then the following occurs, it acts as a regular tree node:
screen shot 2018-11-22 at 6 46 27 pm

Comment (simark) [reply packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx]: Ok so just no icon, makes sense :)

**Pull request merged**